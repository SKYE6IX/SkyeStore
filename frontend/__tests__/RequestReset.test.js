import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import userEvent from '@testing-library/user-event';
import Reset, {
  RESET_MUTATION,
} from '../components/Reset';

const email = 'skye2@mail.com';


const mocks = [
  {
    request: {
      query: RESET_MUTATION,
      variables: { email },
    },
    result: {
      data: { sendUserPasswordResetLink: null },
    },
  },
];

describe('<Reset/>', () => {
  it('renders and matches snapshot', () => {
    const { container } = render(
      <MockedProvider>
        <Reset />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('calls the mutation when submitted', async () => {
    const { container, debug } = render(
      <MockedProvider mock={mocks}>
        <Reset />
      </MockedProvider>
    );
    // type into the email box
    userEvent.type(screen.getByPlaceholderText(/email/i), email);
    // click submit
    userEvent.click(screen.getByText('Request'));
    // const success = await screen.findByText(/Success/i);
    // screen.debug(success)
    // expect(success).toBeInTheDocument();
  });
});