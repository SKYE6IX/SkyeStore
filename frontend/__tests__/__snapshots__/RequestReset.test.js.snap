// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reset/> renders and matches snapshot 1`] = `
<div>
  <form
    class="Form-sc-1xszr8q-0 hVEEdl"
    method="POST"
  >
    <h2>
      Reset Your Password
    </h2>
    <fieldset
      aria-busy="false"
    >
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="Your Email Address"
          type="email"
          value=""
        />
      </label>
      <label
        for="password"
      >
        Password
        <input
          autocomplete="password"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Request
      </button>
    </fieldset>
  </form>
</div>
`;
